### PR TITLE
THEBIGBUGFIX: Refactor context menu code

### PR DESCRIPTION
**Pull request by taylortom**

### CODE REVIEW

File: frontend/src/core/app/contextMenu.js [hunk 1]
```diff
@@ -1,123 +1,101 @@
...
-        title: window.polyglot.t('app.editsettings'),
+    var courseItems = getDefaultItems(['copyID']);
+    var superPerms = ["*/*:create","*/*:read","*/*:update","*/*:delete"];
+    if (Origin.permissions.hasPermissions(superPerms)) {
+      courseItems.push({
+        title: window.polyglot.t('app.export'),
         className: 'context-menu-item',
-        callbackEvent: 'editSettings'
-      },
+        callbackEvent: 'export'
```
Comment (tomgreenfield) [inline frontend/src/core/app/contextMenu.js]: The export event isn't working for me.

File: frontend/src/core/app/views/contextMenuView.js [hunk 2]
```diff
@@ -1,66 +1,52 @@
...
-
-        if (type == 'course') {
-          if (!view.model.isEditable()) {
-            type = 'sharedcourse';
-          }
-        }
-
-        this.onOpenContextMenu(type, e);
-      });
+      this.listenTo(Origin, 'contextMenu:open', this.onOpenContextMenu);
```
Comment (tomgreenfield) [inline frontend/src/core/app/views/contextMenuView.js]: While we're here could we beautify this bit to something like the below?
```js
this.listenTo(Origin, {
  'contextMenu:open': this.onOpenContextMenu,
  'contextMenu:closeContextMenu remove remove:views': this.onCloseContextMenu
}).render();
```

File: frontend/src/core/app/contextMenu.js (hunk 1 above)
Comment (taylortom) [inline frontend/src/core/app/contextMenu.js]: This is a kineo thing, have removed.

File: frontend/src/core/app/contextMenu.js [hunk 3]
```diff
@@ -1,123 +1,92 @@
...
+        title: window.polyglot.t('app.edit'),
         className: 'context-menu-item',
-        callbackEvent: 'edit'
+        callbackEvent: "edit"
       },
       {
         title: window.polyglot.t('app.copy'),
         className: 'context-menu-item',
-        callbackEvent: 'duplicate'
+        callbackEvent: "copy"
```
Comment (tomgreenfield) [inline frontend/src/core/app/contextMenu.js]: The copy option doesn't seem to do anything?

Comment (dancgray) [inline frontend/src/core/app/contextMenu.js]: Also unable to copy a course.

**Comment by taylortom** [comment]
**TODO**: Check this is still relevant, and isn't already in the refactor PR

**Comment by taylortom** [comment]
No longer relevant.

**Pull request closed without merging**